src/main/java/seedu/avo/storage/FileStorage.java

```diff
@@ -0,0 +1,58 @@
+package seedu.avo.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import seedu.avo.exceptions.AvoException;
+
+/**
+ * Represents a file storage
+ * @param <T> type of data stored
+ */
+public class FileStorage<T> extends Storage<T, String> {
+    private final String filePath;
+    private final FileParser<T> parser;
+    public FileStorage(String dirPath, FileParser<T> parser) {
```

Comment:
**Suggestion**: The `dirPath` parameter name is a bit unclear. Could you please think a more descriptive or intuitive name?